### PR TITLE
Search pagination

### PR DESCRIPTION
**Pull request by finestructure**
Merge after #945

### CODE REVIEW

File: Sources/App/Controllers/SearchController.swift [hunk 1]
```diff
@@ -7,7 +7,12 @@ struct SearchController {
 
     func show(req: Request) throws -> EventLoopFuture<HTML> {
         let query = req.query[String.self, at: "query"] ?? ""
-        return API.search(database: req.db, query: query)
+        let page = req.query[Int.self, at: "page"] ?? 1
+        let pageSize = req.query[Int.self, at: "page-size"] ?? Constants.searchPageSize
```
Comment (finestructure) [inline Sources/App/Controllers/SearchController.swift:11]: Not sure about the parameter name `page-size` here, `pageSize`, `page_size`?

File: Sources/App/Core/Constants.swift [hunk 2]
```diff
@@ -31,8 +31,7 @@ enum Constants {
     static let rssFeedMaxItemCount = 100
     static let rssTTL = 60  // minutes
     
-    static let searchLimit = 20
-    static let searchLimitLeeway = 5
```
Comment (finestructure) [inline Sources/App/Core/Constants.swift:35]: I've had to drop the "leeway" mechanism as that was too hard to reconcile with paging.

**Comment by daveverwer** [comment]
This is great 😍

File: Sources/App/Core/Constants.swift (hunk 2 above)
Comment (daveverwer) [reply Sources/App/Core/Constants.swift:35]: Yea, I'd say this is no longer needed.

File: Sources/App/Controllers/SearchController.swift (hunk 1 above)
Comment (daveverwer) [reply Sources/App/Controllers/SearchController.swift:11]: I'd say maybe this doesn't need to be a query string parameter. The page number is certainly necessary, but page size can be fixed in the source. I think once we settle on a value, it won't need to change.

**Review by daveverwer — APPROVED** [review]
Is there one more PR after this to get the pagination controls into the basic search view?

**Comment by finestructure** [comment]
Yes, that'll follow

**Pull request merged**